### ISSUE TITLE
Unexpected symbols decorations for  Python files

### ISSUE THREAD

**Issue by sqs** [sourcegraph/sourcegraph#2473]
When I visit https://sourcegraph.com/github.com/asyncy/cli/-/blob/cli/aliases/config.py (and on 3.1.1 self-hosted), I see decorations on the lines of symbols when I hover in Python files.

1. I did not expect the red background, left border, and after-line text decoration to show up.
1. The after-line text decoration is repeating itself when I hover (each time it adds another after-text decoration with the same text).

This occurs on Python files but not Go files. On Python files it occurs on GitHub via the browser extension too.

![image](https://user-images.githubusercontent.com/1976/53499913-34e0df00-3a5e-11e9-83e5-8772e7fe1304.png)

**Comment by sqs** [comment]
As an immediate fix, it is OK (and preferable) to completely hide the background/border/text decoration. It is not necessary to first fix the problem causing after-line text decoration stacking (that is a bug that's worth fixing, but MUCH lower priority).

**Comment by lguychard** [comment]
@chrismwendt did you mistakenly publish some WIP / local work yesterday?

In the published bundle of `sourcegraph/python`, I found the following code, which doesn't exist in the source of [sourcegraph/sourcegraph-python](https://github.com/sourcegraph/sourcegraph-python):

```javascript
          async hover(n, i) {
            const r = e.app.activeWindow
              ? e.app.activeWindow.visibleViewComponents[0]
              : void 0;
            if (r) {
              const { repo: i, rev: s, path: o } = (0, t.parseUri)(n.uri),
                c = await this.api.search(
                  `repo:${i}@${s} file:${o} type:symbol ^`
                );
              console.log(c),
                r.setDecorations(
                  e.app.createDecorationType(),
                  c.map(t => ({
                    range: new e.Range(t.start.line, 0, t.end.line, 0),
                    border: "solid",
                    borderWidth: "0 0 0 10px",
                    borderColor: "red",
                    backgroundColor: "hsla(0,100%,50%, 0.05)",
                    after: {
                      contentText: `    ${t.symbolName} (of kind ${
                        t.symbolKind
                      })`
                    }
                  }))
                );
            } else console.log("NO EDITOR");
```

I re-published `sourcegraph/python` from the latest master of [sourcegraph/sourcegraph-python](https://github.com/sourcegraph/sourcegraph-python), which fixes this issue.

**Comment by lguychard** [comment]
For reference if you reintroduce these decorations, the text decoration stacking is due to creating a new decoration type each time `setDecorations()` is called, rather than calling `createDecorationType()` once and reusing the returned value.

**Comment by chrismwendt** [comment]
My apologies, I was debugging symbols by showing them as decorations and unintentionally `yarn link`ed my local `basic-code-intel` package into sourcegraph-python locally when I published.

To prevent this in the future, I:

- Added a check to sourcegraph-{python,go} to prevent publishing when there are any linked packages or working tree changes https://github.com/sourcegraph/sourcegraph-python/commit/fcc9f3b45eda2e49da5a523ae09d460ae159f912
- Added a TODO for myself to "Publish all `sourcegraph/{go,python,...}` extensions in CI" in my 3.2 tracking issue https://github.com/sourcegraph/sourcegraph/issues/2452

**Comment by lguychard** [comment]
@chrismwendt sounds good! Do you think it could be worth it to add that check to the extension creator?
